xline/src/client/restore.rs

```diff
@@ -0,0 +1,36 @@
+use std::path::Path;
+
+use engine::{
+    rocksdb_engine::{RocksEngine, RocksSnapshot},
+    StorageEngine,
+};
+use tokio::io::AsyncReadExt;
+
+use crate::{client::errors::ClientError, storage::db::XLINE_TABLES};
+
+/// Restore snapshot to data dir
+/// # Errors
+/// return `ClientError::IoError` if meet io errors
+/// return `ClientError::EngineError` if meet engine errors
+#[inline]
+#[allow(clippy::indexing_slicing)] // safe operation
+pub async fn restore(
+    snapshot_path: impl AsRef<Path>,
+    data_dir: impl AsRef<Path>,
+) -> Result<(), ClientError> {
+    let mut snapshot_f = tokio::fs::File::open(snapshot_path).await?;
+    let tmp_path = format!("/tmp/snapshot-{}", uuid::Uuid::new_v4());
+    let mut rocks_snapshot = RocksSnapshot::new_for_receiving(tmp_path.clone())?;
+    let mut buf = vec![0; 8192];
```

Comment:
Although we can use different buffers with different sizes to store the data read from a snapshot, I still recommend using a unified size to reduce some confusion. Can we use the `MAINTENANCE_SNAPSHOT_CHUNK_SIZE` here? In doing so, we don't need to leave a magic number here or define a constant variable to replace it.

Reply:
Fixed